sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -312,44 +310,53 @@ class AllCombineFn(AccumulatorCombineFn):
   _accumulator_type = AllAccumulator
 
 
-MAX_LONG_10_FOR_LEADING_ZEROS = [19, 18, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15,
-                                 15, 15, 15, 14, 14, 14, 13, 13, 13, 12, 12, 12,
-                                 12, 11, 11, 11, 10, 10, 10, 9, 9, 9, 9, 8, 8,
-                                 8, 7, 7, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 3, 3,
-                                 3, 3, 2, 2, 2, 1, 1, 1, 0, 0, 0]
-
-LONG_SIZE = 64
-
-
-def compare_to(x, y):
-  """return the sign bit of x-y"""
-  if x < y:
-    return 1
-  return 0
+POWER_TEN = [10e-1, 10e0, 10e1, 10e2, 10e3, 10e4, 10e5,
+             10e6, 10e7, 10e8, 10e9, 10e10, 10e11,
+             10e12, 10e13, 10e14, 10e15, 10e16, 10e17,
+             10e18]
 
 
 def get_log10_round_to_floor(element):
-  number_of_leading_zeros = LONG_SIZE - element.bit_length()
-  y = MAX_LONG_10_FOR_LEADING_ZEROS[number_of_leading_zeros]
-  return y - compare_to(element, math.pow(10, y))
+  power = 0
+  while element >= POWER_TEN[power]:
+    power += 1
+  return power - 1
+
+
+class DataflowDistributionCounter(object):
+  """Pure python DataflowDistributionCounter in case Cython not available.
```

Comment:
<!--thread_id:cc_181183985_t; commit:2e80f6f09baf2949f8b46af1b2858a2fa6ee567b; resolved:0-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> You can have that in the file directly, right after the class definition for example:
> 
> ```python
> class SlowDataflowDistributionCounter:
>    ...
> 
> try:
>   from apache_beam.transforms.cy_dataflow_distribution_counter import DataflowDistributionCounter
> except:
>   DistributionCounter = SlowDataflowDistributionCounter
> ```
> 
> Thank you for pointing out that we run both python-only and cython tests.  Please ignore that comment in this case.

<!--section:body-->
From my understanding,  actually I'm doing the same thing now as you suggested.

Reply:
Right, you are doing this implicitly in `DataflowDistributionCounterFn`, but you can make this more explicit by doing the swapping (if possible) right after the slow class definition, and so later you can just do:

```python
class DataflowDistributionCounterFn(AccumulatorCombineFn):
  _accumulator_type = DataflowDistributionCounter
```

Reply:
Thanks! One more thing is, actually I don't think it's necessary to rename as Slow, do you really think it's better with Slow?

Reply:
The advantage is that you will be able to directly refer to the Python version of this class, in the case where the Cython version replaces it.  Will leave this up to you.

Reply:
I think we can do the same things without renaming, please refer to the latest commit https://github.com/apache/beam/pull/5028/commits/88359a8714261269d7bfa8bfdebb855db2edc7b6